bescort.lic

```diff
@@ -289,12 +293,44 @@ class Bescort
     elsif args.hara_polo
       hara_polo(args.mode)
     elsif args.jolas
-      jolas(args.mode)       
+      jolas(args.mode)
+    elsif args.currach
+      currach(args.mode)
     end
   end
 
   private
 
+  def currach(mode)
+    if (mode.include?('aesry') && (Room.current.id == 5555 || Room.current.id == 5557 || Room.current.id == 5558) || (mode.include?('halasa') && Room.current.id == 15863))
+      echo "You're already there silly!"
+      exit    
+    elsif mode == 'aesry'
+      row_direction = 'pier'
+    elsif mode == 'halasa'
+      row_direction = 'rock'
+    else
+      echo("Unrecognized argument: #{mode}")
+      exit
+    end  
+
+    if (Room.current.id == 5555 || Room.current.id == 5557 || Room.current.id == 5558 || Room.current.id == 15863)
+      fput "go currach", "You climb aboard the", "You are aboard", "What are you referring"
```

Comment:
How do you feel about using `bput` and `move` instead of `fput` for all of this new code?

Reply:
Thanks submitted a cleaner version based off your comments.